### DISCUSSION TITLE
Does Tailwind CLI support the nested classes?

### DISCUSSION THREAD

**Discussion by cvkmohan** [tailwindlabs/tailwindcss#7216] [Help] [answered]
Firstly, thanks for a great framework.
I am working with Phoenix, and have integrated Tailwind CLI v 3.0.15 into my application.
However, I have observed that the nested css classes are not working well. Going via the JS bundler approach is the only alternative or am I making any mistake in configuring?

**Comment by jaschaio** [comment]
Same question here, using just the `tailwind cli` nested styles won't work

**Comment by adamwathan** [comment, marked as answer]
The CLI supports custom PostCSS configurations if you run it with the `--postcss` flag. So create a `postcss.config.js` file, install and configure whatever plugins you want, then run the CLI with that flag and you should be good to go.

We don’t include nesting out of the box because most Tailwind users don’t use nesting (myself included) and including it by default would just mean worse build performance for no benefit for those people.

Personally I would discourage using nesting, I think using nesting nudges you to write a lot of custom CSS that is more complicated and harder to maintain than it looks because nesting masks it a bit. You end up with complex, high specificity selectors that often have unintended side effects and can’t be overridden with utility classes.

**Reply by Defite** [reply]
Nesting is useful when you want to rule all your margins and paddings from the parent.

**Reply by fabianwohlfart** [reply]
Thanks. Can wen add this to the docs here?
https://tailwindcss.com/docs/using-with-preprocessors

**Comment by hudri** [comment]
Sadly the postcss option is not really helpful in this use case. As mentioned in #7262, if you want to use nesting or import, you need to have a full Node JS environment with all the PostCSS plugins installed. Which quite contradicts the point of the CLI (especially the standalone CLI).

Especially those environments where NodeJS is not available (e.g. Wordpress on a shared hosting), also have the biggest need for nesting and importing (due editorial content from a database or assets from non-JS frameworks). Discouraged or not, nesting is unavoidable, and the prose plugin is opinionated and tight.

I'd propose to include both, `tailwind/nesting` and `postcss-import`, in the default build of the CLI (at least the CLI build). Maybe disabled by default if this is a problem for compilation, but included in the bundle, so one can enable it by with a config setting.
